servicex/query_core.py

```diff
@@ -320,6 +320,19 @@ def transform_complete(task: Task):
         )
 
         if not cached_record:
+            # Validate the requested code generator only when we are about to
+            # submit a new transform. This avoids a network call when a cached
+            # transform is used.
+            supported_codegens = await self.servicex.get_code_generators_async()
```

Comment:
Shouldn't this make use the client's lazily cached code gen list instead of always going back to the backend for this list?